### ISSUE TITLE
Some Youtube videos do not open

### ISSUE THREAD

**Issue by prog-amateur** [woodruffw/ff2mpv#6]
Hello, I have successfully installed ff2mpv. It works like a charm. 
However, some videos do not open. 

For example this one open fine : 
https://www.youtube.com/watch?v=1Oku4KXWnuI
And this one doesn't open : 
https://www.youtube.com/watch?v=A3VCxAFtP8c

I have Googled and if I understood well, it could be due to youtube-dl, see below : 
https://github.com/mpv-player/mpv/issues/5389

But to be honest, I understand NOTHING about their discussion. Too many codes and technical vocabulary. 

Here is my mpv.conf file : 
```
ontop=yes
border=yes
window-scale=0.2
geometry=100%:100%
ytdl-format=bestvideo[height<=?1080]+bestaudio/best
hwdec-codecs=all
hwdec=method
```

What is your opinion ? Is it related to this youtube-dl ? How can we solve this issue ? 
Thank you very much for your kind help.

**Comment by woodruffw** [comment]
Both of those videos play correctly for me via `ff2mpv`, so it's probably an issue with your MPV configuration, an outdated copy of `youtube-dl`, or locality (YouTube does GeoIP detection, and may refuse to stream a video that's geographically restricted).

Here's my `mpv.conf` -- please see if it works for you:

```
ontop
no-border
force-window
osc=yes

screenshot-format=png
screenshot-template="~/Dropbox/screenshots/%F - %p"

cache-default=50000

alang=jpn
slang=enm,eng

autofit-larger=50%x50%

user-agent="Mozilla/5.0 (X11; Linux; Linux; rv:0.0) Gecko/0 Firefox/0"

ytdl-format='bestvideo[width<=1920][height<=1080]+bestaudio/best'
ytdl-raw-options=yes-playlist=

force-seekable

input-cursor=yes
```

If it still doesn't work (even with that configuration), it's probably either that your copy of `youtube-dl` is outdated (see their website for updating instructions) or that your region is subject to some kind of restriction by YouTube.

**Comment by prog-amateur** [comment]
Thank you very much ! 
I hope to check it tomorrow evening (no access to my laptop currently). 
I will change the config file and see if it is more a youtube-dl issue or my config file.

**Comment by prog-amateur** [comment]
I come back for the feedback with your config file : it doesn't work. 

1. Config file : Not related -> I have used your config file without success. 

2. Youtube restriction : Not related -> I have tested a movie (no explicit content, no copyright) in the same country than I, and I was not able to open it. 

3. Youtube-dl not up-to-date : Not related -> I have updated youtube-dl following `brew install youtube-dl` according to https://rg3.github.io/youtube-dl/download.html

Basically, which package do I need to be sure to use this extension for all movies ? Please, Can we review all of them to be sure I have everything ?

Thank you very much.

**Comment by woodruffw** [comment]
If it's still not working with an up-to-date `youtube-dl`, I'm afraid there's not much I can do. All this extension does is call `mpv`, which in turn just calls `youtube-dl`.

I recommend that you try using `youtube-dl` directly, and seeing if it works:

```bash
$ youtube-dl "https://youtube.com/whatever"
```

and if that doesn't work, report whatever error message it spits out to the youtube-dl team [here](https://github.com/rg3/youtube-dl/issues/new).

**Comment by prog-amateur** [comment]
Hi, I know that you have closed this issue, and thank you for your help. 

I think the issue is related to the way this extension calls youtube-dl / or my laptop because I was able to open both movies with MPV by using : 
`$ youtube-dl "https://youtube.com/whatever"`

Anyway, I will format my laptop (I have tested many softwares/drivers so I need to clean everything with reinstall). I will feedback if the issue is solved after reinstall.

**Comment by woodruffw** [comment]
Out of curiosity (if you don't mind): What OS were you using? I've tested `ff2mpv` very little with OS X, and both `mpv` and `youtube-dl` are probably second-class on it -- there are all kinds of bugs that can pop up on it compared to Linux (where all 3 are better tested).

**Comment by prog-amateur** [comment]
I have installed Xubuntu. But I have made some changes regarding the driver (install proprietary driver, then come back to open-source driver, then change from AMD to Radeon, etc.). 
So, this is why I suspect all those manipulations to be the source of this issue.

**Comment by woodruffw** [comment]
Are you using `brew` with `Xubuntu`? That might be the source of your problem.

The best way to install `youtube-dl` on Xubuntu is probably `pip`:

```
sudo pip install --upgrade youtube-dl
```

Or with a PPA, like https://launchpad.net/~nilarimogard/+archive/ubuntu/webupd8

**Comment by prog-amateur** [comment]
Thank you very much for your feedback. 
`brew` was a mistake from my side, attempting to find some help on the web. I didn't know that brew was for OSX (my brain was starting to mix everything that day). 

So I come back with a very clean install of Xubuntu, no driver setup, no third party software, etc. 
Step-by-step, I have followed this : 

1. MPV
2. ff2mpv
3. mpv.config setup (your setup)
4. change the python path in json file and make the python file executable
5. install youtube-dl via pip
6. try both videos in the first post... 

... and same result. First video open MPV, while second is not possible. 
Very strange, because it is possible to open this specific video via `youtube-dl` in a terminal. I thought that this program just use this `youtube-dl` command, but if it was the case, it would also open the file, what do you think ?

**Comment by woodruffw** [comment]
Yeah, that's very strange. All `ff2mpv` does is call `mpv` directly, and all `mpv` does is call `youtube-dl`:

https://github.com/woodruffw/ff2mpv/blob/dfe44a7ab10312b06766317ca70e716c1a39db74/ff2mpv.py#L5-L12

Do some links still work, while others don't? I can try to do a little bit more local testing.

**Comment by prog-amateur** [comment]
Hi, sorry for my late reply. 

I have a question about your code, I just start to learn Python : 
What does mean lines 8 & 9 ? For instance `--no-terminal`, `--`, etc. 

On my side, I come back with (I hope) useful files : 

1. First one is the MPV log file after failing to play the youtube link (via ff2mpv extension). 
2. Second one is the Terminal process while using the command `$ youtube-dl "https://www.youtube.com/watch?v=A3VCxAFtP8c"

`
[MPV log file after failure.txt](https://github.com/woodruffw/ff2mpv/files/1713523/MPV.log.file.after.failure.txt)
[Terminal process while downloading.txt](https://github.com/woodruffw/ff2mpv/files/1713524/Terminal.process.while.downloading.txt)

Maybe those files contain the solution of my problem. 

Thank you very much.

**Comment by woodruffw** [comment]
`--no-terminal` tells `mpv` not to output anything, since it's being called from another program and not from a terminal. It has no effect on `youtube-dl`:

```
--no-terminal, --terminal
              Disable any use of the terminal  and  stdin/stdout/stderr.  This
              completely silences any message output.
```

`--` tells `mpv` to stop parsing options to treat the next argument as a file. That's there to prevent URLs from adding additional options to `mpv`, which could be a security issue.

Based on the `mpv` log, it looks like a problem with `ytdl_hook`. Have you opened an issue with the `mpv` developers?

**Comment by prog-amateur** [comment]
# OK !!!!!
The issue is solved, great !!!! How ? 
Thanks to God, you bring me to the right place : `ytdl_hook` lead me to this issue `https://github.com/mpv-player/mpv/issues/4748` which lead me to this other issue : `https://github.com/mpv-player/mpv/issues/4728#issuecomment-320725058`. According to mpv developpers, the `ytdl_hook.lua` was the problem. So I had to either put a script, or downgrade youtube-dl. 
I have chosen the easiest way (downgrade) because I didn't know where exactly to put the script. 
To downgrade, just run in a terminal : 
`sudo pip install youtube_dl==2017.07.30.1`

And it woks perfectly !

Thank you so much for your kind help !!!

**Comment by woodruffw** [comment]
Glad to hear it's working!
